app/models/satisfaction_survey_feature_metrics.rb

```diff
@@ -0,0 +1,50 @@
+class SatisfactionSurveyFeatureMetrics
+  include ActionView::Helpers::NumberHelper
+
+  def response_rate(
+    start_time,
+    end_time = Time.zone.now.end_of_day
+  )
+    success_count = 0.0
+    fail_count = 0.0
+    application_forms(start_time, end_time).find_each do |application_form|
+      if application_form.feedback_satisfaction_level? || application_form.feedback_suggestions?
```

Comment:
You could probably implement these conditions in the DB query - e.g. `.where.not(feedback_satisfaction_level: '').where.not(feedback_suggestions: '')`. That way you wouldn't have to load so many model objects into Ruby though you might end up with two queries - one to count the with feedback and one without.

:+1: for using `find_each` versus `each`.

Reply:
Will have a go at refactoring 👍

Reply:
managed to refactor as suggested, please let me know your thoughts :)